### PR TITLE
Omnibus v2

### LINKED ISSUE #2: Anneal gumbel softmax temperature during training
https://github.com/lucidrains/DALLE-pytorch/issues/10#issuecomment-757132197

"wow! temperature feature is awesome! Gradually decreasing it from 5 to 0.05 over 5 epochs and convergence is really fast as well as results look much better!!!"

### PR DESCRIPTION
**Pull request by kingoflolz**
- add temp annealing (closes #2 )
 - adds gradient checkpointing
 - adds bf16
 - add space-to-depth downscaling

### CODE REVIEW

**Review by sdtblck — APPROVED** [review]
:rocket:

**Review by sdtblck — APPROVED** [review]
